### ISSUE TITLE
[Feature request] allowNew as callback to enable validation of new item input.

### ISSUE THREAD

**Issue by Kim-Andersen** [ericgio/react-bootstrap-typeahead#201]
My request is to allow the allowNew prop to take a filter callback, so a new item is only addable when it passes the filter.

Here is what I would like to accomplish:
I'm using typeahead to look up user accounts by name or e-mail. If an account is not found, I'm using the allowNew prop as a feature to invite a user by e-mail. However, at the moment there is no way to validate the input for new items, thus resulting in a new menu item like this: ex: "Invite by email: john@d" - not a valid e-mail.

Basically I would like the add-new option to only appear if the input passes a certain filter.

Otherwise, thank you for a great component - it's very useful and much appreciated.

**Comment by Kim-Andersen** [comment]
Below is some pseudo code to illustrate what I would like to accomplish:
`
<AsyncTypeahead allowNew={(input) => isValidEmail(input)} .../>
`

**Comment by ericgio** [comment]
Interesting idea. The problem I see is where and how to handle validation. For example, I would assume the validation happens when the user tries to select the entry from the menu. What happens for invalid entries? Nothing? Some kind of warning? If the latter, where and how is that displayed?
